### PR TITLE
S3CSI-195: Change length of access keys and update documentation + v2.0.1 bump

### PR DESCRIPTION
**Pull request by anurag4DSB**
- Commit 1: New test case, some enhancements for linter and documentation clarification
- Commit 2: Actual fix
- Commit 3: bump to v2.0.1 + release notes

### CODE REVIEW

File: pkg/driver/node/credentialprovider/provider_test.go [hunk 1]
```diff
@@ -472,3 +473,199 @@ func TestCredentialFallbackEmptySecretData(t *testing.T) {
...
+			name:        "invalid empty secret key",
+			accessKeyID: "AKIAIOSFODNN7EXAMPLE",
+			secretKey:   "",
+			expectError: true,
+			description: "Empty secret key should be rejected",
+		},
+
+		// Edge cases: whitespace handling
+		{
+			name:        "valid access key with leading/trailing spaces (trimmed)",
```
Comment (jonathan-gramain) [inline pkg/driver/node/credentialprovider/provider_test.go]: Feeling nitpicky: you could also add a test with the maximum length allowed AND extra leading/trailing spaces, to check that the length is checked after trimming.

File: pkg/driver/node/credentialprovider/provider_secret.go [hunk 2]
```diff
@@ -18,18 +18,18 @@ const (
...
-	maxAccessKeyIDLen     = 16
-	maxSecretAccessKeyLen = 40
+	// Upper limits support IAM credentials and other providers.
+	maxAccessKeyIDLen     = 128
+	maxSecretAccessKeyLen = 128
 )
 
 /*
-Validation rules (loosened for cloudserver test credentials):
+Validation rules supports IAM credentials, test credentials, and other S3 providers:
```
[the remark this answers is not in the document]
Comment (anurag4DSB) [reply pkg/driver/node/credentialprovider/provider_secret.go]: changed in the latest commit, thanks :)
